framework/include/outputs/PNGOutput.h

```diff
@@ -0,0 +1,73 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// Forward declarations
+class PNGOutput;
+
+template <>
+InputParameters validParams<PNGOutput>();
+
+// pnglib include
+#include <png.h>
+// MOOSE includes
+#include "Output.h"
+
+class PNGOutput : public Output
+{
+public:
+  // Basic constructor.  Takes parameters passed in to create a PNGOutput object.
+  PNGOutput(const InputParameters & parameters);
+
+protected:
+  // Method for assigning color values to the PNG
+  void setRGB(png_byte * rgb, Real selection);
+
+  // Function to create the mesh_function
+  void makeMeshFunc();
+
+  // Function to populate values to the variables used for scaling
+  void calculateRescalingValues();
+
+  // Function that creates the PNG
+  void makePNG();
+
+  // Called to run the functions in this class.
+  virtual void output(const ExecFlagType & /*type*/);
+
+  // Variable to determine the size, or resolution, of the image.
+  Real resolution;
+
+  // Name of file to which the PNG will be written.
+  std::string PNGFile;
```

Comment:
As Cody said: prefix with an underscore. **But** make it lowercase! Symbol names in any scope that begin with an underscore and are followed by a capital letter are reserved for use as implementation macros. We should really get away from adding more of those names.